src/gnc_attitude_estimator.cpp

```diff
@@ -325,14 +325,32 @@ static void ukf(AttitudeEstimatorState &state, AttitudeEstimatorData const &data
   }
 }
 
-/** @fn ukf_m
- *  Update attitude estimator state given a magnetometer reading.
+/** @brief Update attitude estimator state given a magnetometer reading.
  * 
  *  @param[inout] state Attitude filter state.
- *  @param[in]    data  Input sensor data. */
+ *  @param[in]    data  Input sensor data.
+ */
 static void ukf_m(AttitudeEstimatorState &state, AttitudeEstimatorData const &data) {
-  // TODO : Implement this
-  state = AttitudeEstimatorState();
+  ukf(state, data, [](AttitudeEstimatorState &state, AttitudeEstimatorData const &data) -> void {
+    // Calculate Kalman gain
+    UkfMatrix6x3 K;
+    {
+      UkfMatrix3x3 Q, R;
+      lin::qr(lin::ref<3, 3>(state.P_vv, 2, 2), Q, R);
+      lin::backward_sub(R, Q, lin::transpose(Q).eval());
+      K = lin::ref<6, 3>(state.P_xy, 0, 2) * Q;
+    }
+
+    UkfVector3 z_new {
+      data.b_body(0),
+      data.b_body(1),
+      data.b_body(2)
+    };
+
+    // Update the state vector and covariance
+    state.x = state.x_bar + K * (z_new - state.z_bar).eval();
+    state.P = state.P_bar - K * (lin::ref<3, 3>(state.P_vv, 2, 2) * lin::transpose(K)).eval();
+  });
```

Comment:
The magnetometer only update step is essentially identical to the other update step except we are just working with submatrices.